### PR TITLE
Support for json:",omitzero" tag

### PR DESCRIPTION
**Pull request by nsd20463**
This also fixes the 
  X uintptr  json:",omitempty" 
omitempty applied to a zero-value uintptr is treated as empty by the stdlib json.Marshal, but wasn't by easyjson.

### CODE REVIEW

File: tests/data.go [hunk 1]
```diff
@@ -358,6 +358,39 @@ var omitEmptyString = "{" +
...
+
+	StrZ, StrNZ string  `json:",omitzero"`
+	PtrZ, PtrNZ *string `json:",omitzero"`
+
+	IntNZ int `json:"intField,omitzero"`
+	IntZ  int `json:",omitzero"`
+
+	// NOTE: omitzero DOES have effect on non-pointer struct fields.
+	SubZ, SubNZ   SubStruct  `json:",omitzero"`
+	SubPZ, SubPNZ *SubStruct `json:",omitzero"`
```
Comment (neal) [inline tests/data.go]: may be worth also testing pointer values set with zero-value struct/string

Comment (nsd20463) [reply tests/data.go]: sure, why not. done.

**Comment by nsd20463** [comment]
While adding more test cases I realized that combining omitempty and omitzero together generated code which didn't pass `go vet`, so the most recent commit fixes that by treating it as a special case in the encoder generator.

**Review by neal — APPROVED** [review]
LGTM

**Comment by YesYouKenSpace** [comment]
Any way we can merge this in?
